pkg/services/store/httpobjectstore/service.go

```diff
@@ -0,0 +1,369 @@
+package httpobjectstore
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/middleware"
+	"github.com/grafana/grafana/pkg/services/store/kind"
+	"github.com/grafana/grafana/pkg/services/store/object"
+	"github.com/grafana/grafana/pkg/services/store/router"
+	"github.com/grafana/grafana/pkg/util"
+	"github.com/grafana/grafana/pkg/web"
+
+	"github.com/grafana/grafana/pkg/api/response"
+	"github.com/grafana/grafana/pkg/api/routing"
+	"github.com/grafana/grafana/pkg/models"
+)
+
+type HTTPObjectStore interface {
+	// Register HTTP Access to the store
+	RegisterHTTPRoutes(routing.RouteRegister)
+}
+
+type httpObjectStore struct {
+	store  object.ObjectStoreServer
```

Comment:
is the HTTP wrapper interacting with the store as a client?
```suggestion
	store  object.ObjectStoreClient
```

Reply:
🤔 we don't have a client in wire... but likely should

```
no provider found for github.com/grafana/grafana/pkg/services/store/object.ObjectStoreClient
```

This uses the service directly, that has the same API but skips the grpc part